fpga_interchange/arch.cc

```diff
@@ -87,6 +109,43 @@ Arch::Arch(ArchArgs args) : args(args)
     io_port_types.emplace(this->id("$nextpnr_ibuf"));
     io_port_types.emplace(this->id("$nextpnr_obuf"));
     io_port_types.emplace(this->id("$nextpnr_iobuf"));
+
+    if (!this->args.package.empty()) {
+        IdString package = this->id(this->args.package);
+        package_index = -1;
+        for (size_t i = 0; i < chip_info->packages.size(); ++i) {
+            if (IdString(chip_info->packages[i].package) == package) {
+                NPNR_ASSERT(package_index == -1);
+                package_index = i;
+            }
+        }
+
+        if (package_index == -1) {
+            log_error("Could not find package '%s' in chipdb.\n", this->args.package.c_str());
+        }
+    } else {
+        // Default to first package.
```

Comment:
ime, this behaviour has caused significant user confusion before when they forget to specify a package (see #304), it would probably be better to make `--package` required if there is more than one package in the database

Reply:
Sure, done.